server/src/main/java/org/candlepin/hostedtest/HostedTestSubscriptionResource.java

```diff
@@ -128,6 +130,32 @@ protected void createSubscriptionObjects(Subscription subscription) {
                 this.adapter.createProduct(product);
             }
         }
+
+        if (subscription.getProduct() != null && subscription.getProvidedProducts() != null) {
+            ProductData mktProduct = (ProductData) this.adapter.getProduct(subscription.getProduct().getId());
+            Set<ProductData> provided = new HashSet<>();
+
+            for (ProductInfo providedProduct : subscription.getProvidedProducts()) {
+                provided.add((ProductData) this.adapter.getProduct(providedProduct.getId()));
+            }
+
+            mktProduct.setProvidedProducts(provided);
+            this.adapter.productMap.put(mktProduct.getId(), mktProduct);
```

Comment:
Since I updated Subscription to not have a mutator for provided products, and return the product's provided products from its accessor. This block is no longer needed and is completely removed.